File: basics/그리디/1이 될 때까지.py
n, k = map(int, input.split())
result = 0

while n >= k: # n이 k 이상일 때
    remainder = n % k
    if remainder: # 나머지만큼 빼기
        n -= remainder
        result += remainder
    n //= k # n을 k로 나누기
    result += 1

# 마지막으로 남은 수를 1까지 빼기
result += (n - 1)

print(result)

File: basics/그리디/1이 될 때까지.cpp
using namespace std;

int n, k;
int result = 0;

int main() {
    cin >> n >> k;

    while (n >= k) {
      int remainder = n % k;
      if (remainder){
        n -= remainder;
        result += remainder;
      }
      n /= k;
      result += 1;
    }

    result += (n - 1);
    cout << result << '\n';
}
